bug: fix pdp image jutter

sirugh opened the pull request:
## Description

See issue

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here by replacing ISSUE_NUMBER with your actual issue number. -->
<!--- Using the above wording causes Github to automatically close the issue on merge. -->
Closes #1476 .

## Verification Steps

For minicart:

1. Sign in, add an item to cart and open minicart. The image should appear.
2. Clear app cache and disable cache. Refresh the page.
3. Open the cart, you should see the item in cart but the image should be a grey box.

For PDP:
1. Check that images do not cause jutter on the PDP when changing image via clicking thumbnails.

## Screenshots / Screen Captures (if appropriate)

## Proposed Labels for Change Type/Package
<!--- What type of change level would you suggest for this PR? -->
<!--- Major, Minor, or Patch? -->
<!--- See https://magento-research.github.io/pwa-studio/technologies/versioning/ for help -->
- [ ] major (e.g x.0.0 - a breaking change)
- [ ] minor (e.g 0.x.0 - a backwards compatible addition)
- [x] patch (e.g 0.0.x - a bug fix)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have updated the documentation accordingly, if necessary.
- [x] I have added tests to cover my changes, if necessary.

sirugh commented on packages/venia-concept/src/actions/cart/asyncActions.js:
```diff
@@ -350,6 +350,9 @@ export const getCartDetails = (payload = {}) => {
             // then assign its options from the totals subResource
+            // TODO: If we don't have the image in cache we should probably try
+            // to find it some other way otherwise we have no image to display
+            // in the cart and will have to fall back to a placeholder.
```
If a user signs in on a different device after adding an item we have no image to display in the cache. I fixed it by adding some conditional usage in this PR but I also added this TODO because we should probably find a way to re-request a non-cached image for an item added from another device or before a cache clear.

jimbo commented:
How does adding height parameters affect the issue?

sirugh commented:
> How does adding height parameters affect the issue?

The jutter? Because the placeholder is 4:5 and our images are not.

supernova-at commented on packages/venia-concept/src/components/MiniCart/product.js:
```diff
@@ -22,14 +26,26 @@ const Product = props => {
+                      width: imageWidth,
+                      height: imageHeight
+                  })
+                : transparentPlaceholder;
```
Out of curiosity, would we ever need a transparent placeholder that isn't 4x5?

supernova-at replied:
Related, would an alternate fix be to change our transparent placeholder to match the sample data ratio?

sirugh replied:
> Out of curiosity, would we ever need a transparent placeholder that isn't 4x5?

Sure, and we can create/update those when we get to that point. All of the images (except for the landing page category "circle" images) _should_ be 4x5.

> Related, would an alternate fix be to change our transparent placeholder to match the sample data ratio?

I don't think it's worth it right now. @revanth0212 is going to investigate better ways to move forward with images/lazy-loading/etc and I think this may all change anyways.

supernova-at commented on packages/venia-concept/src/components/MiniCart/product.js:
```diff
@@ -22,14 +26,26 @@ const Product = props => {
+                alt={name}
+                classes={{ root: classes.image }}
+                placeholder={transparentPlaceholder}
+                src={src}
```
I'm wondering if we shouldn't just have the `Image` component render the `placeholder` when the `src` image fails to load. We could get rid of the conditional logic above (L32) if the component would take care of that for us.

sirugh replied:
It's already doing that. It'll render a placeholder _until_ the `src` loads, which means if it fails it'll just stay as the placeholder. What the conditional is doing is saying "if there is no source, just render a placeholder AS the source`.

supernova-at commented on packages/venia-concept/src/components/ProductImageCarousel/carousel.css:
```diff
@@ -14,7 +14,7 @@
 
 .currentImage {
-    background-color: rgb(var(--venia-grey));
+    background-color: white;
```
Do we want to create a named CSS variable for this?

sirugh replied:
Sure.

jimbo replied:
(Obviously not for `white`, which is a global constant, but perhaps for how this view uses it.)

supernova-at commented:
✅ Verification steps complete

That said,

> Open the cart, you should see the item in cart but the image should be a grey box.

What is actually happening here? If the image is not cached, shouldn't we be trying to fetch it so we can show it?

sirugh commented on packages/venia-concept/src/components/MiniCart/__tests__/__snapshots__/product.spec.js.snap:
```diff
@@ -2,10 +2,7 @@
-    alt="Unit Test Product"
-    src="path/to/image/source"
-  />
+  path/to/image/source
```
This whole test is unnecessary anymore.

sirugh commented:
> What is actually happening here? If the image is not cached, shouldn't we be trying to fetch it so we can show it?

If a user signs in on a different device after adding an item we have no image to display in the cache. I fixed it by adding some conditional usage in this PR but I also added this TODO because we should probably find a way to re-request a non-cached image for an item added from another device or before a cache clear.

supernova-at commented:
> So yes, we should try, but probably not in the scope of this PR :D

Okay, got it. At any rate, it's an improvement over what currently exists on `develop`, which is an error toast saying "Sorry! An unexpected error occurred. Debug: TypeError36T at joinUrls (/js/client.js)"  😬
